File: Week3/day1/Exercises/ex1.py
# Part I - Review about arrays
people = ["Greg", "Mary", "Devon", "James"]


people.pop(0)


people[people.index("James")] = "Jason"


people.append("YourName")


print("Mary index:", people.index("Mary"))


new_people = people[1:len(people)-1]
print("Copy without Mary and YourName:", new_people)


print("Index of Foo:", people.index("Foo") if "Foo" in people else -1)


last = people[-1]
print("Last person:", last)

# Part II - Loops
print("All people:")
for person in people:
    print(person)

print("Stop after Devon:")
for person in people:
    print(person)
    if person == "Devon":
        break

File: Week3/day1/Exercises/ex1.js
// Part I - Arrays
const people = ["Greg", "Mary", "Devon", "James"];



people.shift(); // removes the first element

people[people.indexOf("James")] = "Jason";



people.push("YourName");



console.log("Index of Mary:", people.indexOf("Mary")); // 0


const newPeople = people.slice(1, people.length - 1);
console.log("New people array:", newPeople); 


console.log("Index of Foo:", people.indexOf("Foo"));


const last = people[people.length - 1];
console.log("Last element:", last); 

// Part II - Loops


console.log("All people:");
for (let i = 0; i < people.length; i++) {
  console.log(people[i]);
}

console.log("Loop stops after Devon:");
for (let i = 0; i < people.length; i++) {
  console.log(people[i]);
  if (people[i] === "Devon") {
    break;
  }
}
